app/code/community/Bolt/Boltpay/Model/ShippingAndTax.php

```diff
@@ -39,18 +39,45 @@ public function applyShippingAddressToQuote( $quote, $shippingAddress ) {
         $region = $directory->getName(); // For region field should be the name not a code.
         $regionId = $directory->getRegionId(); // This is require field for calculation: shipping, shopping price rules and etc.
 
+        $shipping_email = '';
+        if (isset($shippingAddress->email) && !empty($shippingAddress->email)){
```

Comment:
it's overly complicate things :/
Can you use error control operator
http://php.net/manual/en/language.operators.errorcontrol.php